### ISSUE TITLE
Ignoring `RUF102` does not work

### ISSUE THREAD

**Issue by stephenfin** [astral-sh/ruff#23267]
### Summary

This looks like a fun one.

```bash
❯ cat foo.py
x = '123'  # noqa: O103
```

```
❯ cat ruff.toml
[lint]
select = ["E4", "E5", "E7", "E9", "F", "G", "RUF", "S", "UP"]
ignore = ["RUF102"]

❯ ruff check --fix foo.py 
warning: Invalid rule code provided to `# noqa` at foo.py:1: O103
Found 1 error (1 fixed, 0 remaining).
```

I thought I might have the wrong code so I tried `RUF100` through `RUF104`. Curiously it seems I can only disable this warning by specifying *all* 5 codes or using a more generic ignore like `RUF10`.

```
❯ cat ruff.toml
[lint]
select = ["E4", "E5", "E7", "E9", "F", "G", "RUF", "S", "UP"]
ignore = ["RUF100", "RUF101", "RUF102", "RUF103", "RUF104"]

❯ ruff check --fix foo.py 
warning: Invalid rule code provided to `# noqa` at foo/__init__.py:1: O103
All checks passed!
```

You'll note that I still get a warning on the first run though (but not subsequent runs unless I remove `.ruff_cache`). I'd have expected silencing this error to also silence that warning. I'm guessing that warning comes from a different code path that predates `RUF102` and should possibly be removed (?).

Note that I worked around this by using the [`lint.external` setting](https://docs.astral.sh/ruff/settings/#lint_external) to whitelist the non-ruff linter codes.

### Version

ruff 0.15.1

**Comment by stephenfin** [comment]
https://github.com/astral-sh/ruff/issues/23191 looks related though the reporter notes:

> The global suppression under `[tool.ruff.lint]` works though.

so I'm not sure what's going on/what I've missed.

**Comment by ntBre** [comment]
As you noted, that warning is separate from `RUF102`. I agree that we should remove it now that `RUF102` is a lint rule, thank you for the report!

But yes, otherwise `lint.external` is the recommended approach for non-Ruff rule codes, rather than suppressing `RUF102`.

**Comment by ntBre** [comment]
It turns out that there are actually two instances of this warning: one for file-level `noqa` comments and one for line-level comments. RUF102 only covers the line-level case:

```console
❯ ruff check --select RUF102 - <<EOF
# ruff: noqa: X111

1 # noqa: X111
EOF
warning: Invalid rule code provided to `# ruff: noqa` at -:1: X111
warning: Invalid rule code provided to `# noqa` at -:3: X111
RUF102 [*] Invalid rule code in `# noqa`: X111
 --> -:3:3
  |
1 | # ruff: noqa: X111
2 |
3 | 1 # noqa: X111
  |   ^^^^^^^^^^^^
  |
help: Remove the `# noqa` comment
```

I'll go ahead and remove the warning for the covered case, but I think we should also make RUF102 cover file-level suppression comments and then remove that warning as well. We'll probably need to make that a preview change since the rule has already been stabilized.
